src/couch_replicator.erl

```diff
@@ -305,3 +301,77 @@ doc_from_db(RepDb, DocId, UserCtx) ->
          {not_found, _Reason} ->
             {error, not_found}
     end.
+
+
+-spec check_authorization(rep_id(), #user_ctx{}) ->
+    authorized | unauthorized | not_found.
+check_authorization(RepId, #user_ctx{name = Name, roles = Roles}) ->
+    case couch_replicator_scheduler:rep_state(RepId) of
+    #rep{user_ctx = #user_ctx{name = Name}} ->
+        authorized;
+    #rep{user_ctx = #user_ctx{name = _Other}} ->
```

Comment:
That catches the case of an existing replication but a non-matching name. Did you mean it would look better as `#rep{} ->` instead ?